### PR TITLE
refactor(error): show pretty error in planner test

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

After #13264, we print the error sources in PSQL. This PR does the similar stuff for the planner tests.

Also replaced the manual "Cause by" with the source chain. https://github.com/risingwavelabs/risingwave/pull/13348/files#diff-70784a30f4693fe9ecb89ad1b84a60d39562589d2ae378aae9767e0187d64ab3

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/frontend/planner_test/src/lib.rs [hunk 1]
```diff
@@ -253,6 +254,12 @@ pub struct TestCaseResult {
     pub with_config_map: Option<BTreeMap<String, String>>,
 }
 
+macro_rules! report {
+    ($error:expr) => {
+        format!("{:#}", $error.as_report())
```
Comment (xxchan) [inline src/frontend/planner_test/src/lib.rs:259]: Maybe we can add helper methods in `AsReport`: `to_report_string()`, `to_report_string_pretty()`..?

File: src/frontend/planner_test/tests/testdata/output/agg.yaml [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
     values(sum(1));
-  binder_error: |-
-    Bind error: failed to bind expression: sum(1)
+  binder_error: |
+    Failed to bind expression: sum(1)
 
-    Caused by:
-      Invalid input syntax: aggregate functions are not allowed in VALUES
+    Caused by this error:
+      1: Invalid input syntax: aggregate functions are not allowed in VALUES
```
Comment (xxchan) [inline src/frontend/planner_test/tests/testdata/output/agg.yaml:8]: Is this a little verbose? 🤪 (`this error` and `1:`)

File: src/frontend/planner_test/tests/testdata/output/struct_field_access.yaml [hunk 3]
```diff
@@ -49,14 +49,12 @@
...
+    Failed to bind expression: (c).x
 
-    Caused by:
-      Bind error: failed to bind expression: c
-
-    Caused by:
-      internal error: Ambiguous column name: c
+    Caused by these errors (recent errors listed first):
+      1: Failed to bind expression: c
+      2: internal error: Ambiguous column name: c
```
Comment (xxchan) [inline src/frontend/planner_test/tests/testdata/output/struct_field_access.yaml:57]: 😮

File: src/frontend/src/binder/expr/mod.rs [hunk 4]
```diff
@@ -69,10 +69,10 @@ impl Binder {
         //   Feature is not yet implemented: Equal[Int32, Boolean]
```
Comment (xxchan) [inline src/frontend/src/binder/expr/mod.rs:69]: comment above might need to be updated?

File: src/frontend/planner_test/tests/testdata/output/agg.yaml (hunk 2 above)
Comment (BugenZhao) [reply src/frontend/planner_test/tests/testdata/output/agg.yaml:8]: Copied from `snafu`. I'm okay to change that.

Comment (BugenZhao) [reply src/frontend/planner_test/tests/testdata/output/agg.yaml:8]: Welcome for pull requests. 🤣

**Review by xxchan — APPROVED** [review]

**Review by stdrc — APPROVED** [review]

**Pull request merged**